components/brave_shields/browser/ad_block_pref_service.cc

```diff
@@ -56,6 +61,10 @@ AdBlockPrefService::AdBlockPrefService(AdBlockService* ad_block_service,
   OnPreferenceChanged(prefs::kFBEmbedControlType);
   OnPreferenceChanged(prefs::kTwitterEmbedControlType);
   OnPreferenceChanged(prefs::kLinkedInEmbedControlType);
+
+  if (is_regular_profile) {
+    MaybeRecordInitialShieldsSettings(prefs, host_content_settings_map);
```

Comment:
I don't really follow why do we need to pass all the stuff (HostContentSettingsMap, is_regular_profile) to this service (why this one?) just to record P3A. Am I missing something important about this specific location? Maybe we could do that around `BraveProfileManager` that already has some P3A initialization?

Reply:
now that you point it out, i agree. moved the function call to `browser/profiles/profile_util.cc`